### DISCUSSION TITLE
Getting 404 error in log in App Router in production env

### DISCUSSION THREAD

**Discussion by Senbonzakura1234** [vercel/next.js#55531] [App Router] [answered]
My app (deploy at https://sophie-dex.vercel.app) at every route making a request to url https://sophie-dex.vercel.app/?_rsc=6tjqs which return 404 everytime. Try to access the url directly but no beaking error happened. 

Only happen in production env;

Here is my [source code](https://github.com/Senbonzakura1234/sophie-dex)

**Comment by devjiwonchoi** [comment]
Seems all good to me, could you share steps to repro?

**Reply by Senbonzakura1234** [reply]
@devjiwonchoi i just simply navigate to the url https://sophie-dex.vercel.app/ and i receive 404 in console log. Never happened in local env. so i'm not sure how to reproduce either. You can try download my source code and deploy it on vercel yourself to reproduce, i had push a change to allow no env variable check since this error surely not relate to my database or my api fetching.

**Reply by Senbonzakura1234** [reply]
@devjiwonchoi okay so i narrowed it down the cause is that the component next/link Link to Home page (path: '/') try to prefetch home page and failed. I tried to recreate my Link component at [deploy](https://test-rsc-404-error.vercel.app/) - [repo](https://github.com/Senbonzakura1234/test-rsc-404-error) with minimal setup but failed to reproduce the error. I think the issue lies in the content of my [home page](https://github.com/Senbonzakura1234/sophie-dex/tree/main/src/app/(home)) itself, something inside it makes the Link component can't prefetch it.

**Reply by Senbonzakura1234** [reply]
@devjiwonchoi right now my only solution is to [disable prefetch for homepage](https://github.com/Senbonzakura1234/sophie-dex/pull/2/files)

**Comment by Senbonzakura1234** [comment]
@devjiwonchoi okay so i found the root cause.
it's the i18n option in my next.config.mjs file left off from when i migrate the codebase from page router to app router. 
```ts
const config = {
	i18n: { locales: ['en'], defaultLocale: 'en' },
	images: { remotePatterns: [{ protocol: 'https', hostname: 'avatars.githubusercontent.com', port: '' }] },
	experimental: { typedRoutes: true, mdxRs: true },
};
```

**Reply by Senbonzakura1234** [reply, marked as answer]
@devjiwonchoi tried with `i18n: { locales: ['en-US'], defaultLocale: 'en-US' },` still not worked. The only way to completely remove the error is to remove this i18n option for good.

**Comment by wannesmatthys** [comment]
I have this exact problem. Can't remove i18n because I need it... did you figure this out?

**Reply by Senbonzakura1234** [reply]
@wannesmatthys i think you should [create an issue ticket](https://github.com/vercel/next.js/issues/new/choose) for this, this is a bug no doubts. Just remember to mention this thread in the ticket, it could help people debug.

**Reply by wannesmatthys** [reply]
Created an issue: https://github.com/vercel/next.js/issues/57741
